lib/streamlit/delta_generator.py

```diff
@@ -528,11 +528,6 @@ def _enqueue(
                 "call your fragment function inside a `with st.sidebar` context manager."
             )
 
-        # Warn if we're called from within a legacy @st.cache function
-        legacy_caching.maybe_show_cached_st_function_warning(dg, delta_type)
-        # Warn if we're called from within @st.memo or @st.singleton
-        caching.maybe_show_cached_st_function_warning(dg, delta_type)
```

Comment:
it looks like you moved this logic into the individual elements via `check_widget_usage()`, right? 
It might make sense to leave a comment in the PR description to elaborate on this a little bit

Reply:
Added a bit more to the PR description